crates/pipeline-manager/tests/integration_test.rs

```diff
@@ -1633,6 +1711,226 @@ not_a_number,true,ΑαΒβΓγΔδ
         .await;
 }
 
+/// Test completion tokens with a pipeline that has no output connectors.
+#[actix_web::test]
+#[serial]
+async fn completion_tokens() {
+    let config = setup_for_pipeline(Some("test")).await;
+
+    create_and_deploy_test_pipeline(
+        &config,
+        "create table t1(c1 integer, c2 bool, c3 varchar) with ('materialized' = 'true'); create materialized view v1 as select * from t1;",
+    )
+        .await;
+
+    // Start the pipeline
+    let response = config.post_no_body("/v0/pipelines/test/start").await;
+    assert_eq!(response.status(), StatusCode::ACCEPTED);
+    config
+        .wait_for_deployment_status(
+            "test",
+            PipelineStatus::Running,
+            Duration::from_millis(1_000),
+        )
+        .await;
+
+    for i in 0..1000 {
+        let token: CompletionTokenResponse = config
+            .post_json(
+                "/v0/pipelines/test/ingress/T1?format=json&update_format=raw",
+                format!(r#"{{"c1": {i}, "c2": true}}"#),
+            )
+            .await
+            .json()
+            .await
+            .unwrap();
+        println!("Iteration {i}, token: {}", token.token);
```

Comment:
does this generate lots of outputs in the log?

Reply:
only if the test fails, in which case these outputs are useful